### PR TITLE
C4-255 Inheritance Mode

### PR DESCRIPTION
**Pull request by willronchetti**
- Ports Burak's inheritance mode code into CGAP, added in the ingestion listener when it is building variant samples
- Makes additional mapping table updates
- Fixes VCF file status patch behavior
- Fixes some potential exception raising code that should not raise exception

### CODE REVIEW

File: src/encoded/tests/data/workbook-inserts/file_processed.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  },
+  {
+    "file_format": "vcf_gz",
+    "project": "12a92962-8265-4fc0-b2f8-cf14f05db58b",
+    "institution": "hms-dbmi",
+    "file_ingestion_status": "N/A",
+    "uuid": "a0661ad2-099e-4092-b9e3-c35db70a17c1",
+    "accession": "GAPFIZ654321"
+  }
+]
```
Comment (netsettler) [inline src/encoded/tests/data/workbook-inserts/file_processed.json:18]: Missing newline.

File: src/encoded/inheritance_mode.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+                if role not in d:
+                    return []
+
+        if (cls.check_if_label_exists(cls.GENOTYPE_LABEL_DOT, genotype_labels) or
+                cls.is_multiallelic_site(genotypes.values()) or
+                cls.check_if_label_exists(cls.GENOTYPE_LABEL_SEX_INCONSISTENT, genotype_labels)):
+            return []
+
+        # De novo strong candidate
+        if novoPP > 0.9:
```
Comment (netsettler) [inline src/encoded/inheritance_mode.py:198]: Can we name these constants, please? The nature of these thresholds and why the particular values were chosen should be documented. Also, if the same constant occurs elsewhere, it should be clear whether it's for the same or an unrelated reason/purpose.

File: src/encoded/commands/variant_table_intake.py [hunk 3]
```diff
@@ -490,14 +490,47 @@ def generate_variant_sample_schema(self, sample_props, cols, facs, variant_cols,
...
             'type': 'string',
             'enum': ['proband', 'father', 'mother', 'brother', 'sister', 'sibling', 'half-brother', 'half-sister',
                      'half-sibling', 'wife', 'husband', 'grandson', 'granddaughter', 'grandchild',
                      'grandmother', 'family-in-law', 'extended-family', 'not linked']
         }
+        schema['properties']['samplegeno']['items']['properties']['samplegeno_sex'] = {  # noqa structure is there
+            'title': 'Sex',
+            'description': 'Sex of the donor of this sample ID',
+            'type': 'string',
+            'enum': ['M', 'F', 'U']  # XXX: what others should be included?
```
Comment (netsettler) [inline src/encoded/commands/variant_table_intake.py:532]: https://www.hl7.org/fhir/codesystem-gender-identity.html

File: src/encoded/ingestion_listener.py [hunk 4]
```diff
@@ -54,6 +56,25 @@ def ingestion_status(context, request):
...
+    kwargs = {
+        'environ': request.environ,
+        'method': 'GET',
+        'content_type': 'application/json'
+    }
+    subreq = Request.blank('/' + uuid, **kwargs)
+    resp = request.invoke_subrequest(subreq, use_tweens=True)
+    if isinstance(resp, HTTPMovedPermanently):  # if we hit a redirect, follow it
+        subreq = Request.blank(resp.location, **kwargs)
+        resp = request.invoke_subrequest(subreq, use_tweens=True)
```
Comment (netsettler) [inline src/encoded/ingestion_listener.py:72]: What determines when `use_tweens` should be `True` or `False`?

File: src/encoded/commands/ingestion.py [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import os
 import argparse
 import logging
-from ..commands.variant_ingestion import run_variant_table_intake, run_ingest_vcf
-from ..commands.gene_ingestion import run_gene_table_intake, run_ingest_genes
+from encoded.commands.variant_ingestion import run_variant_table_intake, run_ingest_vcf
```
[the remark this answers is not in the document]
Comment (willronchetti) [reply src/encoded/commands/ingestion.py:4]: If I use relative imports here this command cannot be run from the top level of the repo (where `production.ini` is for example).

**Pull request merged**